docs/source/executing-operations/defer-support.mdx

```diff
@@ -0,0 +1,170 @@
+
+```graphql
+query GetPersonAndFriends($personId: ID!) {
+  person(id: $personId) {
+    # Basic fields (fast)
+    id
+    name {
+      firstName
+      lastName
+    }
+
+    # highlight-start
+    # Friend fields (slower)
+    ... on User @defer {
+      friends {
+        id
+      }
+    }
+    # highlight-end
+  }
+}
+```
+
+When responding to the above query, a GraphQL server can "chunk" its response, starting by returning only the "basic fields". Later, it can return an additional chunk that includes the "friend fields".
+
+As shown, deferred fields are always contained within a GraphQL fragment, and the `@defer` directive is applied to that fragment (_not_ to the individual fields).
+
+## Using `@defer` with the router
+
+> To use `@defer` successfully with the Apollo Router, a client's GraphQL library must _also_ support the directive by handling "chunked" responses correctly.
+
+The Apollo Router supports deferring the following fields in your schema:
+
+* Fields that are nested anywhere within an [entity type](#entity-fields)
+* Root fields of the [`Query` type](#fields-of-query)
+
+In most supergraphs, this represents a large percentage of all fields in the schema. [See an example of unsupported fields.](#unsupported-field-example)
+
+### Entity fields
+
+The Apollo Router supports using `@defer` with fields and nested subfields of [entity types](/federation/entities/) in your federated supergraph. Entities are object types with at least one `@key`, which specifies the fields that can uniquely identify any given instance of that type.
+
+Here's an example subgraph schema that defines a `User` entity:
+
+```graphql title="Users subgraph"
+type User @key(fields: "id") {
+  id: ID!
+  name: Name!
+  friends: [User!]!
+}
+
+type Name {
+  firstName: String!
+  lastName: String!
+}
+
+type Query {
+  person(id: ID!): User
+}
+```
+
+Because the `User` type is an entity, a client query can apply `@defer` to any fragment that's nested somewhere within that type (at any depth).
+
+Here are two example queries:
+
+```graphql
+query QueryA($personId: ID!) {
+  person(id: $personId) {
+    id
+    # highlight-start
+    ... on User @defer {
+      name {
+        firstName
+        lastName
+      }
+    }
+    # highlight-end
+  }
+}
+
+query QueryB($personId: ID!) {
+  person(id: $personId) {
+    id
+    name {
+      firstName
+      # highlight-start
+      ... on Name @defer {
+        lastName
+      }
+      # highlight-end
+    }
+  }
+}
+```
+
+### Fields of `Query`
+
+The Apollo Router also supports using `@defer` with root fields of your supergraph's `Query` type:
+
+```graphql
+query PartiallyDeferredQuery {
+  products {
+    id
+  }
+  # highlight-start
+  ... on Query @defer {
+    users {
+      id
+    }
+  }
+  # highlight-end
+}
+```
+
+You _cannot_ use `@defer` with root fields of the `Mutation` type.
+
+### Unsupported field example
+
+Let's say our router's supergraph schema includes the following:
```

Comment:
I think using "our router's supergraph schema" is a tad confusing here if we're trying to highlight that `Book` is not an entity type. Because if we've talking of the supergraph API schema, then `@key` never appear there anyway, so all type will "look" as if they weren't entity if that's what you look at.

And if by "supergraph schema" we meant "the actually composed supergraph we use internally", then I would discourage about taking about _that_ in user documentation in general. Because imo, the supergraph (in that sense) should largely be considered an implementation detail. What make sense to a end use is:
1. subgraphs schemas.
2. their supergraph API schema.

And besides, even the raw supergraph schema don't have `@key` either, it has some `@join__*` directives instead (and that's exactly why the raw supergraph should imo be considered an implementation detail: those `@join` directives aren't meant to be something our average has to know exists).

Tl;r, we should be referencing to a subgraph schema here instead.

Reply:
👍 Using subgraph schema now